### ISSUE TITLE
Support `AssignExpr` where possible

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1137]
We use a lot of assignment related checks in our code. We fetch the number of variables in a function, calculate names overlap, etc. `python3.8` introduced new way to assign things with `:=`
And while I dislike the general approach, we need to be sure that everything works correctly even when users have these bad structure inside their code.

However, we still do not allow to use `:=` inside the code with this rule: #34 
This is only for better user experience.

Blocked by:
- #1136 
- #1135

**Comment by sobolevn** [comment]
We also don't know currently where this should be added. As the first step - we can find these places and create a list of them.

**Comment by sobolevn** [comment]
One interesting case:

```python
if x := 1:
    print(x)
```
